### PR TITLE
Lead gen email popup design

### PR DESCRIPTION
**Pull request by TMcMeans**

### CODE REVIEW

File: apps/marketing/components/Layout/index.js [hunk 1]
```diff
@@ -25,6 +28,19 @@ const Layout = ({
       hasBanner = false
     }
   }
+
+  const [isOpen, openModal] = useState(false)
+
+  useEffect(() => {
+    window.setTimeout(() => {
+      openModal(_.get(popupCopy, 'h1[0]') === 'true')
```
Comment (craigspaeth) [inline apps/marketing/components/Layout/index.js]: 👍 nice!

File: apps/marketing/components/Layout/index.js [hunk 2]
```diff
@@ -38,6 +54,7 @@ const Layout = ({
         locationGroups={locationGroups}
         hasBanner={hasBanner}
       />
+      <PopupModal isOpen={isOpen} closeModal={closeModal} {...popupCopy} />
```
Comment (craigspaeth) [inline apps/marketing/components/Layout/index.js]: Nice simple use of hooks here!

File: apps/marketing/components/PopupModal/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const PopupModal = ({ isOpen, closeModal, h1, images, p }) => {
+  const handleClick = () => {
+    console.log('form submitted')
+  }
+
+  return (
+    <Modal isOpen={isOpen} closeModal={closeModal}>
+      <div className={cx('inner')}>
+        <div className={cx('left-copy')}>
+          <img src={_.get(images, '0.url')} />
```
Comment (craigspaeth) [inline apps/marketing/components/PopupModal/index.js]: 👍 solid use of content modules and `_.get`!

**Comment by craigspaeth** [comment]
👏 Looking good so far! No constructive comments—well done!

File: apps/marketing/components/PopupModal/index.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  const [emailInput] = [useRef()]
+
+  const [state, setState] = useState({
+    successMsg: '',
+    errorMsg: ''
+  })
+
+  const handleSubmit = async e => {
+    e.preventDefault()
+    const gql = new GraphQLClient(process.env.APP_URL + '/api')
```
[the remark this answers is not in the document]
Comment (TMcMeans) [reply apps/marketing/components/PopupModal/index.js]: Using existing envar for app URL

**Comment by TMcMeans** [comment]
This PR is technically done, there are just some CSS tweaks that I gotta finish & responsive styling to add

**Comment by TMcMeans** [comment]
Ready for merging!

File: apps/api/models/patient/mapper.js [hunk 5]
```diff
@@ -4,7 +4,8 @@ import {
   toCliniko,
   toModel,
   referralToTreatmentNote,
-  sfWaitlistNoteToTreatmentNote
+  sfWaitlistNoteToTreatmentNote,
+  leadGenNoteToTreatmentNote
```
Comment (craigspaeth) [inline apps/api/models/patient/mapper.js]: Not for this PR, but we'll probably want to think about pulling the treatment note out as a first-class model to generalize this and DRY up the code.

File: apps/marketing/components/Layout/index.js [hunk 6]
```diff
@@ -25,6 +29,24 @@ const Layout = ({
...
   }
+
+  const [isOpen, openModal] = useState(false)
+
+  useEffect(() => {
+    const cookie = Cookies.get('marketingModalClosed')
+
+    if (!cookie) {
+      window.setTimeout(() => {
+        openModal(_.get(popupCopy, 'h1[0]') === 'true')
```
Comment (craigspaeth) [inline apps/marketing/components/Layout/index.js]: 👍 nice work on this opening logic.

File: apps/marketing/components/PopupModal/index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          input: {
+            name: emailInput.current.value,
+            email: emailInput.current.value,
+            leadGenNote: {
+              body: 'Marketing lead gen'
+            }
+          }
+        }
+      )
+      .catch(error => {
```
Comment (craigspaeth) [inline apps/marketing/components/PopupModal/index.js]: 👍 thanks for remembering the unhappy path and handling errors! FWIW You can use `try { await gql(...) } catch (err) { err = }` here as well

File: apps/marketing/components/PopupModal/index.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        err = _.get(error, 'response.errors[0].message')
```
Comment (craigspaeth) [inline apps/marketing/components/PopupModal/index.js]: Nit-pick: `err` isn't used outside of here so you can just use `const err =` instead.

File: apps/marketing/components/PopupModal/popup-modal.css [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    font-weight: bold;
+    font-size: 0.8em;
+    cursor: pointer;
+    transition: background-color var(--transition-out);
+}
+
+.join-btn:hover {
+    background-color: var(--dark-yellow-color);
+    transition-duration: var(--transition-in);
+}
```
Comment (craigspaeth) [inline apps/marketing/components/PopupModal/popup-modal.css]: Wrangled this CSS file like a boss!

**Comment by craigspaeth** [comment]
:clap: Well done! None of my comments are blockers, so mergin' Thank you 🙌

**Pull request merged**